supply/spec/uploader_spec.rb

```diff
@@ -203,22 +203,69 @@ def find_obbs
       end
     end
 
-    describe '#perform_upload' do
-      let(:client) { double('client') }
-      let(:config) { { apk: 'some/path/app.apk' } }
+    RSpec::Matchers.define(:same_localizedtext_as) do |expected_arr|
+      match do |actual_arr|
+        actual_arr.sort_by(&:language).zip(expected_arr.sort_by(&:language)).map do |actual_localization, expected_localization|
+          actual_localization.language == expected_localization.language && actual_localization.text == expected_localization.text
+        end.reduce(:&)
+      end
+    end
 
-      before do
-        Supply.config = config
-        allow(Supply::Client).to receive(:make_from_config).and_return(client)
-        allow(client).to receive(:upload_apk).with(config[:apk]).and_return(1) # newly uploaded version code
-        allow(client).to receive(:begin_edit).and_return(nil)
-        allow(client).to receive(:commit_current_edit!).and_return(nil)
+    shared_examples 'run supply to upload metadata' do
+      describe '#perform_upload' do
+        subject(:release) { OpenStruct.new({ version_codes: version_codes }) }
+        let(:client) { double('client') }
+        let(:config) { { apk_paths: version_codes.map { |v_code| "some/path/app-v#{v_code}.apk" }, metadata_path: 'supply/spec/fixtures/metadata/android', track: 'track-name' } }
+
+        before do
+          Supply.config = config
+          allow(Supply::Client).to receive(:make_from_config).and_return(client)
+          version_codes.each do |version_code|
+            allow(client).to receive(:upload_apk).with("some/path/app-v#{version_code}.apk").and_return(version_code) # newly uploaded version code
+          end
+          allow(client).to receive(:upload_changelogs).and_return(nil)
+          allow(client).to receive(:tracks).with('track-name').and_return([OpenStruct.new({ releases: [ release ] })])
```

Comment:
Not tested, but I think you could use an object double like `double('tracks', releases: [release])` instead of using `OpenStruct.new`? Which would me more idiomatic of rspec tests practices I think?